pvlib/pvsystem.py

```diff
@@ -898,6 +1092,281 @@ def __repr__(self):
+        to lookup albedo.
+
+    surface_type : None or string, default None
+        The ground surface type. See ``irradiance.SURFACE_ALBEDOS``
+        for valid values.
+
+    module : None or string, default None
+        The model name of the modules.
+        May be used to look up the module_parameters dictionary
+        via some other method.
+
+    module_type : None or string, default 'glass_polymer'
+         Describes the module's construction. Valid strings are 'glass_polymer'
+         and 'glass_glass'. Used for cell and module temperature calculations.
+
+    module_parameters : None, dict or Series, default None
+        Module parameters as defined by the SAPM, CEC, or other.
+
+    temperature_model_parameters : None, dict or Series, default None.
+        Temperature model parameters as defined by the SAPM, Pvsyst, or other.
+
+    modules_per_string: int, default 1
+        Number of modules per string in the array.
+
+    strings: int, default 1
+        Number of strings in the array.
+
+    racking_model : None or string, default 'open_rack'
+        Valid strings are 'open_rack', 'close_mount', and 'insulated_back'.
+        Used to identify a parameter set for the SAPM cell temperature model.
+
+    """
+
+    def __init__(self,
+                 surface_tilt=0, surface_azimuth=180,
+                 albedo=None, surface_type=None,
+                 module=None, module_type=None,
+                 module_parameters=None,
+                 temperature_model_parameters=None,
+                 modules_per_string=1, strings=1,
+                 racking_model=None, name=None):
+        self.surface_tilt = surface_tilt
+        self.surface_azimuth = surface_azimuth
+
+        self.surface_type = surface_type
+        if albedo is None:
+            self.albedo = irradiance.SURFACE_ALBEDOS.get(surface_type, 0.25)
+        else:
+            self.albedo = albedo
+
+        self.module = module
+        if module_parameters is None:
+            self.module_parameters = {}
+        else:
+            self.module_parameters = module_parameters
+
+        self.module_type = module_type
+        self.racking_model = racking_model
+
+        self.strings = strings
+        self.modules_per_string = modules_per_string
+
+        if temperature_model_parameters is None:
+            self.temperature_model_parameters = \
+                self._infer_temperature_model_params()
+        else:
+            self.temperature_model_parameters = temperature_model_parameters
+
+        self.name = name
+
+    def __repr__(self):
+        attrs = ['name', 'surface_tilt', 'surface_azimuth', 'module',
+                 'albedo', 'racking_model', 'module_type',
+                 'temperature_model_parameters',
+                 'strings', 'modules_per_string']
+        return 'Array:\n  ' + '\n  '.join(
+            f'{attr}: {getattr(self, attr)}' for attr in attrs
+        )
+
+    def _infer_temperature_model_params(self):
```

Comment:
Commenting here in case we should discuss: is there a concern about moving these private methods from `PVSystem` to `Array`  If anyone was using them as 

```
pv = PVSystem()
pv._infer_temperature_model_params
```

They would now be at `pv.arrays._infer_temperature_model_params`

Reply:
IMO you use private methods at your own peril. Since they are private they they may be subject to unannounced and dramatic changes. That said, if these methods have come in to common usage we should consider making them public and providing some kind of backwards compatibility.